common/beerocks/bcl/source/son/son_wireless_utils.cpp

```diff
@@ -869,3 +869,15 @@ std::list<uint8_t> wireless_utils::string_to_wsc_oper_class(const std::string &o
         return {};
     }
 }
+
+bool wireless_utils::channel_validate(const uint8_t &channel)
+{
+    for (auto oper_class : operating_classes_list) {
+        auto ch = oper_class.second.channels.find(channel);
+
+        if (ch != oper_class.second.channels.end()) {
+            return true;
+        }
+    }
+    return false;
```

Comment:
That's not a very efficient implementation for this operation.
If we don't care about the operating classes, maybe we need to have an `std::set` with all the "valid" channels and search it instead of iterating over all the operating classes.

Reply:
> That's not a very efficient implementation for this operation.
> If we don't care about the operating classes, maybe we need to have an std::set with all the "valid" channels and search it instead of iterating over all the operating classes.

IMO, we shouldn't bother with efficiency too much.

There are a dozen different ways that the operating class table is accessed. If we would keep a copy of the table for each access pattern, it would be very hard to maintain all those copies. For how many microseconds gain? We're talking about a tiny table with just about a 100 entries here...